### PR TITLE
issue #92 - re-enable JDBC search tests

### PR DESCRIPTION
**Pull request by lmsurpre**
1. correct issue where all JDBC search test tenants had the same extensions file (the date one); while some of the content is similar, each tenant has its own unique configuration (where all defined search parameters are of the type that matches the tenant name)
2. introduce extension-search-parameters-valuetypes.json files for the remaining JDBC search test tenants (one per search param type)
3. add null checking and include root cause in logging messages while loading these value types
4. move the remaining JDBC Search test json files to fhir-examples
5. update DerbyInitializer to use the new schema behavior from Robin
6. fix bug in FHIRPersistenceJDBCNormalizedImpl for detecting integer searches
7. enable the JDBC Search tests in the default JDBC TestNG suite

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-examples/src/main/resources/json/ibm/basic/BasicReference.json [hunk 1]
```diff
@@ -49,25 +49,25 @@
       }
     },
     {
-      "url": "http://example.org/ReferenceId",
+      "url": "http://example.org/Reference-id",
```
Comment (prb112) [inline fhir-examples/src/main/resources/json/ibm/basic/BasicReference.json]: is this a pattern I should follow if I run into extensions?

File: fhir-examples/src/main/resources/json/ibm/basic/BasicToken.json [hunk 2]
```diff
@@ -57,15 +54,13 @@
       }
     },
     {
-      "__comment": "If the system is present, and there is no code, then this is understood to mean that there is no suitable code in the system in which to represent the code.",
```
Comment (prb112) [inline fhir-examples/src/main/resources/json/ibm/basic/BasicToken.json]: are these removed as no longer functional or the details are in the test?

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/DerbyInitializer.java [hunk 3]
```diff
@@ -116,14 +113,7 @@ public void bootstrapDb(boolean reset) throws FHIRPersistenceDBConnectException,
         }
         else {
             System.out.println("Bootstrapping database");
-            final String url = DERBY_TRANSLATOR.getUrl(dbProps);
-            try (Connection connection = DriverManager.getConnection(url + ";create=true")) {
```
Comment (prb112) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/DerbyInitializer.java]: removing this makes create=false.  I think in this case don't you want to set to true? I can't think of the exact reason we hit this before, but just worth checking that create=true is specified in DerbyFhirDatabase.

File: fhir-examples/src/main/resources/json/ibm/basic/BasicReference.json (hunk 1 above)
Comment (lmsurpre) [reply fhir-examples/src/main/resources/json/ibm/basic/BasicReference.json]: doesn't really matter; just needs to match whats in the corresponding SearchParameter in extension-search-parameters.json

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/DerbyInitializer.java (hunk 3 above)
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/DerbyInitializer.java]: `new DerbyFhirDatabase()` is how Robin was instantiating the database R4JDBCExamplesTest, so I was just trying to be consistent with that.

File: fhir-examples/src/main/resources/json/ibm/basic/BasicToken.json (hunk 2 above)
Comment (lmsurpre) [reply fhir-examples/src/main/resources/json/ibm/basic/BasicToken.json]: in past (DSTU2 timeframe) our parser was lenient by default and ignored unexpected elements.  now its strict by default and so these cause issues.
john tells me there is still "special" support for a field called `fhir_comments` but indicated that he might like to remove that in the future and so I stayed away.

**Pull request merged**